src/network/networkd-ndisc.c

```diff
@@ -1942,6 +1943,11 @@ static int ndisc_router_process_encrypted_dns(Link *link, sd_ndisc_router *rt) {
                 return 0;
         }
 
+        if (set_size(link->ndisc_dnr) >= NDISC_ENCRYPTED_DNS_MAX) {
+                log_link_debug(link, "Too many Encrypted DNS records received. Only first %u records will be used.", NDISC_ENCRYPTED_DNS_MAX);
```

Comment:
I can see the log message:
```sh
Nov 05 05:38:16 default systemd-networkd[87]: host0: Too many Encrypted DNS records received. Only first 64 records will be used.
...
```
The functions processing the DNSSL and RDNSS options use `log_link_warning` though with the `logged_about_too_many` logic.

Reply:
(I'm not sure how important that is though)

Reply:
Ok sounds reasonable. I'll make it a warning for consistency.

Reply:
FWIW looking at
```sh
Nov 05 05:51:44 systemd-networkd[87]: host0: Too many RDNSS records per link. Only first 64 records will be used.
Nov 05 05:51:44 systemd-networkd[87]: host0: Too many RDNSS records per link. Only first 64 records will be used.
Nov 05 05:52:37 systemd-networkd[87]: host0: Too many RDNSS records per link. Only first 64 records will be used.
Nov 05 05:52:37 systemd-networkd[87]: host0: Too many RDNSS records per link. Only first 64 records will be used.
```
it doesn't seem that `logged_about_too_many` helps too much. It's probably a leftover. (It's unrelated to this PR though)